### ISSUE TITLE
New rule: Decimal with float argument

### ISSUE THREAD

**Issue by kbaskett248** [astral-sh/ruff#11840]
I'm interested in adding a new rule to check for `Decimal` being called with a float literal argument. For example:

```py
# Don't
d = Decimal(1.234)
# Do
d = Decimal('1.234')
```

This seems like a generally useful check. I searched for open issues and PRs involving `Decimal` and didn't find any mention of this type of check. I'm happy to work on a PR, but wanted to make sure this check makes sense in `ruff`. Please let me know if I'm missing something or there's any reason this check wouldn't be accepted.

Thanks!

**Comment by charliermarsh** [comment]
I don't _think_ we have a check for this. Can you say more on why it's something to be avoided?

**Comment by mayanyax** [comment]
Consider:

```
>>> Decimal(0.3) == Decimal(0.1) + Decimal(0.1) + Decimal(0.1)
False
>>> Decimal('0.3') == Decimal('0.1') + Decimal('0.1') + Decimal('0.1')
True
```

In the first example you basically creating an exact representation of the floating point approximation to 0.1. To my mind that is almost certainly not what is wanted as it seems to defeat the purpose of using Decimal in the first place.

**Comment by kbaskett248** [comment]
Yeah, @mayanyax is exactly right. Passing a `float` literal could introduce imprecision, which is exactly what we're hoping to avoid.

**Comment by Matthieu-LAURENT39** [comment]
this is a footgun i keep forgetting, and i would definitely enjoy seeing this rule implemented.

**Comment by AlexWaygood** [comment]
Fixed by #12909
